tests/GameRulesTest.cs

```diff
@@ -0,0 +1,246 @@
+        }
+        
+        [Test]
+        public void ReturnsTrueForDiagonalWinWhenFirstDiagonalIsO()
+        {
+            _board.SetSpot(1, _oMarker);
+            _board.SetSpot(5, _oMarker);
+            _board.SetSpot(9, _oMarker);
+            Assert.True(_gameRules.DiagonalWin(_board, _oMarker));
+        }
+        
+        [Test]
+        public void ReturnsTrueForDiagonalWinWhenSecondDiagonalIsX()
+        {
+            _board.SetSpot(3, _xMarker);
+            _board.SetSpot(5, _xMarker);
+            _board.SetSpot(7, _xMarker);
+            Assert.True(_gameRules.DiagonalWin(_board, _xMarker));
+        }
+        
+        [Test]
+        public void ReturnsFalseForDiagonalWinWhenBoardIsEmptyOrTie()
+        {
+            Assert.False(_gameRules.DiagonalWin(_board, _xMarker)); 
+            Assert.False(_gameRules.DiagonalWin(_board, _oMarker));
+            Assert.False(_gameRules.DiagonalWin(_tieBoard, _oMarker));
+            Assert.False(_gameRules.DiagonalWin(_tieBoard, _xMarker));
+        }
+        
+        [Test]
+        public void ReturnsTrueForWinWhenFirstRowIsX()
+        {
+            _board.SetSpot(1, _xMarker);
+            _board.SetSpot(2, _xMarker);
+            _board.SetSpot(3, _xMarker);
+            Assert.True(_gameRules.Win(_board, _xMarker));
+        }
+        
+        [Test]
+        public void ReturnsTrueForWinWhenThirdColumnIsO()
+        {
+            _board.SetSpot(3, _oMarker);
+            _board.SetSpot(6, _oMarker);
+            _board.SetSpot(9, _oMarker);
+            Assert.True(_gameRules.Win(_board, _oMarker));
+        }
+        
+        [Test]
+        public void ReturnsTrueForWinWhenSecondDiagonalIsX()
+        {
+            _board.SetSpot(3, _xMarker);
+            _board.SetSpot(5, _xMarker);
+            _board.SetSpot(7, _xMarker);
+            Assert.True(_gameRules.Win(_board, _xMarker));
+        }
+        
+        [Test]
+        public void ReturnsFalseForWinWhenBoardIsEmptyOrTie()
+        {
+            Assert.False(_gameRules.Win(_board, _xMarker)); 
+            Assert.False(_gameRules.Win(_board, _oMarker));
+            Assert.False(_gameRules.Win(_tieBoard, _oMarker));
+            Assert.False(_gameRules.Win(_tieBoard, _xMarker));
+        }
+
+        [Test]
+        public void ReturnsFalseForWinWhenBoardIsNotEmptyButNotCompletelyFull()
+        {
+            _board.SetSpot(1, _xMarker);
+            _board.SetSpot(2, _xMarker);
+            _board.SetSpot(3, _oMarker);
+            _board.SetSpot(5, _oMarker);
+            _board.SetSpot(6, _oMarker);
+            _board.SetSpot(9, _xMarker);
+            
+            Assert.False(_gameRules.Win(_board, _xMarker)); 
+            Assert.False(_gameRules.Win(_board, _oMarker));
+        }
+        
+        [Test]
+        public void ReturnsTrueForTieWhenBoardIsTie()
+        {
+            Assert.True(_gameRules.Tie(_tieBoard));
+            Assert.True(_gameRules.Tie(_tieBoard));

```

Comment:
Looks like all the `.Tie` assertions are doubled. I guess this used to take `_xMarker` and `_oMarker`? 😃

Reply:
yes :) removed duplicate line in https://github.com/olgaboiar/ttt_csharp/pull/1/commits/0df81a04b911b94ff8be42adcdd621eaecb5bf6c